tasks.c

```diff
@@ -3551,20 +3553,20 @@ static portTASK_FUNCTION( prvIdleTask, pvParameters )
              * because the scheduler is suspended. */
             eReturn = eAbortSleep;
         }
-        else
-        {
-            /* If all the tasks are in the suspended list (which might mean they
-             * have an infinite block time rather than actually being suspended)
-             * then it is safe to turn all clocks off and just wait for external
-             * interrupts. */
-            if( listCURRENT_LIST_LENGTH( &xSuspendedTaskList ) == ( uxCurrentNumberOfTasks - uxNonApplicationTasks ) )
+
+        #if ( INCLUDE_vTaskSuspend == 1 )
+            else if( listCURRENT_LIST_LENGTH( &xSuspendedTaskList ) == ( uxCurrentNumberOfTasks - uxNonApplicationTasks ) )

```

Comment:
Readability - Wonder if this #if directive, the blank line above it, and the extra indent (per uncrustify) are causing a readability issue.

Would you prefer:
```
    }
    else if( taskARE_TASKS_ALL_SUSPENDED() )
    {
```

where `taskARE_TASKS_ALL_SUSPENDED()` is defined conditionally to `pdFALSE` when `INCLUDE_vTaskSuspend` is 0.